lib/netext/httpext/transport.go

```diff
@@ -164,8 +168,35 @@ func (t *transport) measureAndEmitMetrics(unfReq *unfinishedRequest) *finishedRe
 			tags["ip"] = ip
 		}
 	}
+	var failed float64
+	if t.responseCallback != nil {
+		var statusCode int
+		if unfReq.response != nil {
+			statusCode = unfReq.response.StatusCode
+		}
+		expected := t.responseCallback(statusCode)
+		if !expected {
+			failed = 1
+		}
+
+		if enabledTags.Has(stats.TagExpectedResponse) {
+			tags[stats.TagExpectedResponse.String()] = strconv.FormatBool(expected)
+		}
+	}
 
-	trail.SaveSamples(stats.IntoSampleTags(&tags))
+	finalTags := stats.IntoSampleTags(&tags)
+	trail.SaveSamples(finalTags)
+	if t.responseCallback != nil {
+		trail.Failed.SetValid(true)
+		if failed == 1 {
+			trail.Failed.Bool = true
+		}
+		trail.Samples = append(trail.Samples,
+			stats.Sample{
```

Comment:
I worry a bit that this will cause a slice expansion every time, given that we have expected response and response callbacks enabled by default. So it may be worth it to initialize the `trail.Samples` slice with a cap +1 in `SaveSamples()`?

Reply:
yeah ... I wondered about that, and whether to do with a ... boolean/int flag saying to add 1/the number :thinking: